src/DotNetLightning.Core/Channel/ChannelTypes.fs

```diff
@@ -397,3 +397,70 @@ type ChannelState =
             | ErrFundingLost _
             | ErrFundingTimeOut _
             | ErrInformationLeak _ -> Abnormal
+
+        member this.Commitments: Option<Commitments> =
+            match this with
+            | WaitForInitInternal
+            | WaitForOpenChannel _
+            | WaitForAcceptChannel _
+            | WaitForFundingCreated _
+            | WaitForFundingSigned _ -> None
+            | WaitForFundingConfirmed data -> Some (data :> IHasCommitments).Commitments
+            | WaitForFundingLocked data -> Some (data :> IHasCommitments).Commitments
+            | Normal data -> Some (data :> IHasCommitments).Commitments
+            | Shutdown data -> Some (data :> IHasCommitments).Commitments
+            | Negotiating data -> Some (data :> IHasCommitments).Commitments
+            | Closing data -> Some (data :> IHasCommitments).Commitments
+            | Closed _
+            | Offline _
+            | Syncing _
+            | ErrFundingLost _
+            | ErrFundingTimeOut _
+            | ErrInformationLeak _ -> None
+
+        member this.SpendableBalance: Option<LNMoney> =
+            match this.Commitments with
+            | None -> None
+            | Some commitments ->
+                let remoteCommit =
+                    match commitments.RemoteNextCommitInfo with
+                    | RemoteNextCommitInfo.Waiting info -> info.NextRemoteCommit
+                    | RemoteNextCommitInfo.Revoked _info -> commitments.RemoteCommit
+                let reducedRes =
+                    remoteCommit.Spec.Reduce(
+                        commitments.RemoteChanges.ACKed,
+                        commitments.LocalChanges.Proposed
+                    )
+                let reduced =
+                    match reducedRes with
+                    | Error err ->
+                        failwithf
+                            "reducing commit failed even though we have not proposed any changes\
+                            error: %A"
+                            err
+                    | Ok reduced -> reduced
+                let fees =
+                    if commitments.LocalParams.IsFunder then
+                        Transactions.commitTxFee commitments.RemoteParams.DustLimitSatoshis reduced
+                        |> LNMoney.FromMoney
+                    else
+                        LNMoney.Zero
+                let channelReserve =
+                    commitments.RemoteParams.ChannelReserveSatoshis
+                    |> LNMoney.FromMoney
+                let totalBalance = reduced.ToRemote
+                let untrimmedSpendableBalance = totalBalance - channelReserve - fees
+                let htlcSuccessFee =
+                    reduced.FeeRatePerKw.ToFee(Transactions.Constants.HTLC_SUCCESS_WEIGHT)
+                    |> LNMoney.FromMoney
+                let htlcFee =
+                    reduced.FeeRatePerKw.ToFee(Transactions.Constants.COMMITMENT_TX_WEIGHT_PER_HTLC)
```

Comment:
@canndrew no need for parenthesis here either

Reply:
Fixed.